clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -540,6 +540,9 @@ void CodeGenFunction::EmitOpenCLKernelMetadata(const FunctionDecl *FD,
   if (!FD->hasAttr<OpenCLKernelAttr>())
     return;
 
+  if (getLangOpts().SYCLIsDevice)
+    Fn->addFnAttr("module-id", Fn->getParent()->getModuleIdentifier());
```

Comment:
Would it make sense to have something more specific, such as `sycl-module-id` instead?

Reply:
Seems reasonable to me, done.